uO_scrape.py

```diff
@@ -137,13 +138,163 @@ def get_courses(link):
 		comp = comp.split(':', 1)[-1].strip()
 		comp = [x.strip() for x in comp.split('/')[-1].split(',')]
 		#getting the prerequisites from after the colon in the sentence
-		pre = pre.split(':', 1)[-1].strip()
-		#extracting dependencies from the prerequisite list
-		dep = extract_codes('.'.join([x for x in pre.split('.') if ('cannot' not in x) and ('Must' not in x)]))
-		dep = set(dep).difference({code})
-		courses.append([code, title, credits, desc, comp, pre, dep])
+		pre = Prereq(pre)
+		dep = None
+		# pre = pre.split(':', 1)[-1].strip()
+		# #extracting dependencies from the prerequisite list
+		# dep = extract_codes('.'.join([x for x in pre.split('.') if ('cannot' not in x) and ('Must' not in x)]))
+		# dep = set(dep).difference({code})
+		courses.append([code, title, credits, desc, comp, pre.prereqs, dep])
```

Comment:
I think that's definitely a good idea, and I thought about leaving it as an object, but as per your suspicion when outputting as a json file (didn't try csv), it doesn't save the object to a file in any meaningful way.

My thought was at least for the moment this would not break outputting to json. I tend to prioritize not breaking existing features before thinking about future features, although since this is a project in early development getting partial functionality in the way we want to have it long term may be preferable.

I'll let you choose which you prefer.

Reply:
No worries, that makes sense!  I'll note it down as a future task to work on!